client/src/components/UploadPortfolioItem.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useState} from 'react'
+import FileUpload from './FileUpload'
+import '../styles/upload-portfolio-item.scss'
+import {addPortfolio} from '../Backend'
+
+interface Props {handle: string}
+
+
+const UploadPortfolioItem: React.FC <Props> = ({handle}: Props) => {
+  const [description, setDescription] = useState<string>('')
+  const [previewFile, setPreviewFile] = useState<string|null>('')
+
+  async function handleSubmit() {
+    const portfolioItem = {
+      description: description,
+      file: previewFile,
+      handle: handle
+    }
+  
+    try{
+      await addPortfolio(portfolioItem)
+      console.log('Portfolio submitted')
+    } catch(err) {
+      alert(err.message)
+    }
+  }
+
+  async function setFile(file: File | null) {
+    let fileBase64: Promise<string|null>;
+    if (file) {
+      fileBase64 = new Promise(resolve => {
+        const reader = new FileReader()
+        reader.addEventListener('load', () => {
+          resolve(reader.result && reader.result.toString())
+        })
+        reader.readAsDataURL(file)
+      })
+    } else {
+      fileBase64 = Promise.resolve(null)
+    }
+    let filePreview = await fileBase64
+    setPreviewFile(filePreview || null)
```

Comment:
I think you can just do this:
```suggestion
    setPreviewFile(await fileBase64)
```